zipline/algorithm.py

```diff
@@ -110,73 +110,78 @@
 
 
 class TradingAlgorithm(object):
-    """
-    Base class for trading algorithms. Inherit and overload
-    initialize() and handle_data(data).
-
-    A new algorithm could look like this:
-    ```
-    from zipline.api import order, symbol
-
-    def initialize(context):
-        context.sid = symbol('AAPL')
-        context.amount = 100
-
-    def handle_data(context, data):
-        sid = context.sid
-        amount = context.amount
-        order(sid, amount)
-    ```
-    To then to run this algorithm pass these functions to
-    TradingAlgorithm:
-
-    my_algo = TradingAlgorithm(initialize, handle_data)
-    stats = my_algo.run(data)
-
+    """A class that represents a trading strategy and parameters to execute
+    the strategy.
+
+    Parameters
+    ----------
+    *args
+        Forwarded ``initialize``.
+    initialize : callable[context -> None, optional
+        Function that is called at the start of the simulation to
+        setup the initial context.
+    handle_data : callable[(context, data) -> None], optional
+        Function called on every bar. This is where most logic should be
+        implemented.
+    before_trading_start : callable[(context, data) -> None], optional
+        Function that is called before any bars have been processed each
+        day.
+    analyze : callable[(context, DataFrame) -> None], optional
+        Function that is called at the end of the backtest. This is passed
+        the context and the performance results for the backtest.
+    script : str, optional
+        Algoscript that contains the definitions for the four API
+        functions and any supporting code.
+    namespace : dict, optional
+        The namespace to execute the algoscript in. By default this is an
+        empty namespace that will include only python built ins.
+    algo_filename : str, optional
+        The filename for the algoscript. This will be used in exception
+        tracebacks. default: '<string>'.
+    data_frequency : {'daily', 'minute'}, optional
```

Comment:
the numpy convention for an optional set of arguments is that the first element is the default